src/style/style_layer/fill_extrusion_style_layer.js

```diff
@@ -81,34 +81,40 @@ export function getIntersectionDistance(projectedQueryGeometry: Array<Point>, pr
         // Check whether points are coincident and use other points if they are.
         let i = 0;
         const a = projectedFace[i++];
-        let b, c;
+        let b;
         while (!b || a.equals(b)) {
             b = projectedFace[i++];
             if (!b) return Infinity;
         }
-        while (!c || a.equals(c) || b.equals(c)) {
-            c = projectedFace[i++];
-            if (!c) return Infinity;
-        }
 
-        const p = projectedQueryGeometry[0];
+        // Loop until point `c` is not colinear with points `a` and `b`.
+        for (; i < projectedFace.length; i++) {
+            const c = projectedFace[i];
+
+            const p = projectedQueryGeometry[0];
+
+            const ab = b.sub(a);
+            const ac = c.sub(a);
+            const ap = p.sub(a);
 
-        const ab = b.sub(a);
-        const ac = c.sub(a);
-        const ap = p.sub(a);
+            const dotABAB = dot(ab, ab);
+            const dotABAC = dot(ab, ac);
+            const dotACAC = dot(ac, ac);
+            const dotAPAB = dot(ap, ab);
+            const dotAPAC = dot(ap, ac);
+            const denom = dotABAB * dotACAC - dotABAC * dotABAC;
```

Comment:
can this be replaced by `ab.cross(ac)`? you could also use that to check for co-linerarity, right?

Reply:
The objects are the `Point` class which doesn't have `.cross()` (I think that is in one of the gl-matrix classes). If it was already there this would be probably cleaner but I'm not sure it's worth adding at this point

> you could also use that to check for co-linerarity, right?

At first I checked for collinearity directly but because of precision stuff we'd need to pick a threshold for what we count as colinear. At first I hit a case where my colinear check said they were not colinear enough but the whole method still failed to produce a valid distance. I think checking the output should be more reliable because it avoids that gap

Reply:
Fair enough, makes sense to me!